File: sqlite-android-studio-plugin/src/main/java/com.alecstrong.sqlite.android/lang/SqliteFindUsagesHandlerFactory.java
package com.alecstrong.sqlite.android.lang;

import com.alecstrong.sqlite.android.psi.ColumnNameElement;
import com.alecstrong.sqlite.android.util.SqliteRenameUtil;
import com.intellij.find.findUsages.FindUsagesHandler;
import com.intellij.find.findUsages.FindUsagesHandlerFactory;
import com.intellij.find.findUsages.JavaFindUsagesHandler;
import com.intellij.find.findUsages.JavaFindUsagesHandlerFactory;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class SqliteFindUsagesHandlerFactory extends FindUsagesHandlerFactory {
  @Override public boolean canFindUsages(@NotNull PsiElement element) {
    return element instanceof ColumnNameElement;
  }

  @Nullable @Override
  public FindUsagesHandler createFindUsagesHandler(@NotNull final PsiElement element,
      boolean forHighlightUsages) {
    if (element instanceof ColumnNameElement) {
      return new JavaFindUsagesHandler(element,
          SqliteRenameUtil.getSecondaryElements((ColumnNameElement) element,
              (SqliteFile) element.getContainingFile()),
          JavaFindUsagesHandlerFactory.getInstance(element.getProject()));
    }

    return null;
  }
}

File: sqlite-android-studio-plugin/src/main/kotlin/com/alecstrong/sqlite/android/lang/SqliteFindUsagesHandlerFactory.kt
package com.alecstrong.sqlite.android.lang

import com.alecstrong.sqlite.android.psi.SqliteElement.ColumnNameElement
import com.alecstrong.sqlite.android.util.getSecondaryElements
import com.intellij.find.findUsages.FindUsagesHandlerFactory
import com.intellij.find.findUsages.JavaFindUsagesHandler
import com.intellij.find.findUsages.JavaFindUsagesHandlerFactory
import com.intellij.psi.PsiElement

class SqliteFindUsagesHandlerFactory : FindUsagesHandlerFactory() {
  override fun canFindUsages(element: PsiElement) = element is ColumnNameElement

  override fun createFindUsagesHandler(element: PsiElement, forHighlightUsages: Boolean) =
      when (element) {
        is ColumnNameElement -> JavaFindUsagesHandler(element, element.getSecondaryElements(),
            JavaFindUsagesHandlerFactory.getInstance(element.getProject()))
        else -> null
      }
}
